### PR TITLE
Update build scripts to create a Netlify deploy

### PR DESCRIPTION
**Pull request by Paul-Hebert**
This PR adds the ability for a number of our sites to be hosted on Netlify at https://portable-patterns.netlify.app/

There were a few changes required for this:
- Update the various build scripts to point to a central location
- Update our build scripts (This could be fine-tuned further in the future, but for now it gets the job done)

I also updated the app titles so their context was clearer and added a new vanilla HTML landing page that has some info and links to the other apps. Feel free to suggest changes to this content, or edit it in a follow-up PR. (We can also change the Netlify URL if you'd like)

Right now Netlify is hosting Angular, Ember, React, Vue, and Storybook. These were the easiest to make static. If there are other apps you'd like to see hosted, or if you'd like to make the various components into packages let me know and I can make a follow-up PR. 

Once this is merged we should switch the Netlify site to deploy from `master`

### CODE REVIEW

File: copy-vanilla-html.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+var copydir = require('copy-dir');
+
+// Copy vanilla html to dist
+copydir('vanilla-html', 'dist', {});
```
Comment (Paul-Hebert) [inline copy-vanilla-html.js:4]: TODO Newline

File: package.json [hunk 2]
```diff
@@ -4,12 +4,25 @@
...
+    "build:components:stencil": "cd stencil-components && npm run build && cd ../",
+    "build:components:react-components": "cd react-components && npm run build && cd ../",
+    "build:components:angular-components": "cd angular-components && npm run build && cd ../",
+    "build:components:vue-components": "cd vue-components && npm run build && cd ../",
+
+    "build:app:angular": "npm ci --prefix angular-app && npm run build --prefix angular-app",
+    "build:app:ember": "npm ci --prefix ember-app && npm run build --prefix ember-app",
+    "build:app:react": "npm ci --prefix react-app && npm run build --prefix react-app",
+    "build:app:storybook": "npm ci --prefix storybook && npm run build --prefix storybook",
+    "build:app:vue": "npm ci --prefix vue-app && npm run build --prefix vue-app",
```
Comment (Paul-Hebert) [inline package.json:18]: TODO: Make `install:app` script

File: react-app/.env.production [hunk 3]
```diff
@@ -0,0 +1 @@
+PUBLIC_URL=/react/
```
Comment (Paul-Hebert) [inline react-app/.env.production:1]: TODO: Newline

File: vanilla-html/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+public
```
Comment (Paul-Hebert) [inline vanilla-html/.gitignore:1]: TODO: Newline

File: vanilla-html/index.html [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      <c4-button tag="a" href="/angular">Angular</c4-button>
+      <c4-button tag="a" href="/ember">Ember</c4-button>
+      <c4-button tag="a" href="/storybook">Storybook</c4-button>
+      <c4-button tag="a" href="/react">React</c4-button>
+      <c4-button tag="a" href="/vue">Vue</c4-button>
+    </div>
+  </c4-container>
+</body>
+
+</html>
```
Comment (Paul-Hebert) [inline vanilla-html/index.html:74]: TODO: Newline

File: package.json (hunk 2 above)
Comment (Paul-Hebert) [reply package.json:18]: This turned into a rabbit hole. Punting

**Review by grigs — APPROVED** [review]
👍

File: package.json (hunk 2 above)
Comment (gerardo-rodriguez) [inline package.json:18]: Nit: I see an opportunity to make the script names a bit more consistent. 

Pattern 1: `build:<type>:<framework>` (e.g. `build:app:vue`, `build:components:stencil`)
Pattern 2: `build:<type>:<framework>-<type>` (e.g. `build:components:vue-components`) 

Might it make sense to settle on one or the other? My preference is the `build:<type>:<framework>` version.

This is not a merge blocker. If this is too big of a hassle, feel free to ignore. 🙂

**Review by gerardo-rodriguez — APPROVED** [review]
Looks good, as far as I can tell. 👍

File: package.json (hunk 2 above)
Comment (Paul-Hebert) [reply package.json:18]: Hmm, good question. Let me take a look and see how much work this would be to adjust.

Comment (Paul-Hebert) [reply package.json]: Done! https://github.com/cloudfour/tpp-research-monorepo/pull/14/commits/58d6261b4b26833c5960a6491407c9ab57ffd5e2

**Review by gerardo-rodriguez — APPROVED** [review]

**Pull request merged**